librz/cons/grep.c

```diff
@@ -86,6 +86,25 @@ static void parse_grep_expression(const char *str) {
 	RzConsGrep *grep = &cons->context->grep;
 	sorted_column = 0;
 	bool first = true;
+
+	// setup cons->context->grep.icase according to cons->grep_icase
+	if (cons->grep_icase == CASE_SMART) {
+		// smartcase - when the search term is all lowercase, ignore the case,
+		// instead if the search term is uppercase or a mix, do a case-sensitive search.
+		bool has_upper = false;
+		const char *grep_str = str;
+		while (*grep_str) {
+			if (isupper(*grep_str)) {
+				has_upper = true;
```

Comment:
this becomes true the first time it finds an upper case. maybe we can break it and make it faster?

Reply:
Thank you. Add `break` already.